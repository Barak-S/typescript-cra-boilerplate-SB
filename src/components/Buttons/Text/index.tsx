import React, { FC, MouseEvent } from 'react';
import { colors, ms, ClassNameProps, StyleProps, Styles, useHover } from 'styles';

interface Props extends StyleProps, ClassNameProps {
  size?: number;
  href?: string;
  disabled?: boolean;
  onClick?: () => void;
}

export const TextButton: FC<Props> = ({ className, style, children, href, disabled, size = 14, onClick }) => {
  const handleClick = (e: MouseEvent<HTMLAnchorElement>) => {
    if (disabled) {
      return;
    }
    if (!href && onClick) {
      e.preventDefault();
      onClick();
    }
  };

  const { hover, hoverProps } = useHover();
  const styles = getStyles(size);

  return (
    <a
      className={className}
      style={ms(styles.container, disabled && styles.disabled, hover && styles.hover, style)}
      href={href}
      onClick={handleClick}
      {...hoverProps}
    >
      {children}
    </a>
  );
};

const getStyles = (size: number): Styles => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    color: colors.coolBlueTwo,
    boxSizing: 'border-box',
    textDecoration: 'none',
    whiteSpace: 'nowrap',
    transition: 'all .3s ease-in-out',
    cursor: 'pointer',
    fontSize: `${size}px`,
  },
  hover: {
    textDecoration: 'underline',
  },
  disabled: {
    color: colors.greyish,
    cursor: 'not-allowed',
    textDecoration: 'none',
  },
});

export type TextButtonProps = Props;
export default TextButton;
