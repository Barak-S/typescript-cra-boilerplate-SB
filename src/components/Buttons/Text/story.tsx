import { View } from 'components/Common';
import React from 'react';
import { sbAutoDetectActionProps, sbStyles, Story, StoryMeta } from 'utils';

import TextButton, { TextButtonProps as Props } from '.';

export default ((): StoryMeta<Props> => ({
  title: 'components/Buttons/Text',
  component: TextButton,
  args: {
    children: 'Text button',
  },
  argTypes: {},
  parameters: {
    layout: 'centered',
    actions: { ...sbAutoDetectActionProps },
  },
}))();

export const Basic: Story<Props> = args => (
  <View column>
    <TextButton {...args} style={sbStyles.mb10} />
    <TextButton {...args} disabled />
  </View>
);
