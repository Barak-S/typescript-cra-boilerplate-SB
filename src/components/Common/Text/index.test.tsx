import { shallow } from 'enzyme';
import React from 'react';

import Text from '.';

describe('<Text />', () => {
  it('should render', () => {
    const component = shallow(<Text style={{ height: 10 }}>{'Hello'}</Text>);
    expect(component).toMatchSnapshot('hello');
  });
});
