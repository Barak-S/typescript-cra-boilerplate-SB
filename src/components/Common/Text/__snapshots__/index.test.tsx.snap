// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render: hello 1`] = `
<span
  style={
    Object {
      "height": 10,
    }
  }
>
  Hello
</span>
`;
